examples/common_spaces.jl

```diff
@@ -0,0 +1,87 @@
+using ClimaCore: Geometry, Domains, Meshes, Topologies, Spaces
+
+abstract type Space{FT} end
+
+Base.@kwdef struct PeriodicLine{FT} <: Space{FT}
+    xmax::FT
+    xelem::Int
+    npoly::Int
+end
+
+Base.@kwdef struct PeriodicRectangle{FT} <: Space{FT}
+    xmax::FT
+    ymax::FT
+    xelem::Int
+    yelem::Int
+    npoly::Int
+end
+
+Base.@kwdef struct CubedSphere{FT} <: Space{FT}
```

Comment:
@dennisYatunin , I know this has been merged already. But I did not have a chance to go through this before and I have questions. Why the need to redefine the CubedSphere struct here?